### PR TITLE
replace deprecated code in favor of Java 7 core and apache commons libraries

### PR DESCRIPTION
**Pull request by elharo**

### CODE REVIEW

**Comment by elharo** [comment]
This is a common misconception. Instance fields are **not** shared between test methods.

**Comment by dennisl** [comment]
My bad. Then we are good to go!

File: src/test/java/org/apache/maven/shared/filtering/DefaultMavenResourcesFilteringTest.java [hunk 1]
```diff
@@ -758,20 +752,20 @@ private void createTestDataStructure( File sourceDirectory )
...
+        FileUtils.write( new File( emptyDirectory, ".gitignore" ), "# .gitignore file", "UTF-8" );
 
         File emptyDirectoryChild = new File( sourceDirectory, "empty-directory-child" );
         emptyDirectory.mkdirs();
 
         File emptyDirectoryChildEmptyChild = new File( emptyDirectoryChild, "empty-child" );
         emptyDirectoryChildEmptyChild.mkdirs();
 
-        FileUtils.fileWrite( new File( emptyDirectoryChildEmptyChild, ".gitignore" ), "UTF-8", "# .gitignore file" );
+        FileUtils.write( new File( emptyDirectoryChildEmptyChild, ".gitignore" ), "# .gitignore file", "UTF-8" );
```
Comment (olamy) [inline src/test/java/org/apache/maven/shared/filtering/DefaultMavenResourcesFilteringTest.java:768]: simply using Files.write could avoid external dependencies

Comment (elharo) [reply src/test/java/org/apache/maven/shared/filtering/DefaultMavenResourcesFilteringTest.java:768]: not quite a 1:1 replacement. What we really need is Files.writeString from Java 11. We'd also need to change the File objects to Path objects.   I'll leave this change for a separate PR if someone wants to take it.

**Comment by elharo** [comment]
java.io.FileNotFoundException: File does not exist: /home/jenkins/jenkins-slave/workspace/maven-box_maven-filtering_i912/m/target/sourceTestGitIgnoreFile
	at org.apache.maven.shared.filtering.DefaultMavenResourcesFilteringTest.testShouldReturnGitIgnoreFiles(DefaultMavenResourcesFilteringTest.java:677)

**Comment by elharo** [comment]
Passes locally though.

**Pull request merged**